spec/railtie_spec.rb

```diff
@@ -27,4 +27,35 @@ def index
       expect(ActiveRecord::Relation.ancestors).to include(Readyset::RelationExtension)
     end
   end
+  describe 'readyset.query_annotator' do
+    context 'when query_log_tags_enabled is true' do
+      before do
+        config = Rails.configuration.active_record
+        allow(config).to receive(:query_log_tags_enabled).and_return(true)
+      end
+
+      it 'adds a query_log_tag for routing to Readyset' do
+        # Exercise
+        # Here we are assuming that Readyset::QueryAnnotator.routing_to_readyset? returns a value
+        allow(Readyset::QueryAnnotator).to receive(:routing_to_readyset?).and_return(true)
+
+        # Verify
+        expect(Rails.configuration.active_record.query_log_tags).to include(
+          a_hash_including(routed_to_readyset?: an_instance_of(Proc))
+        )
+      end
+    end
+
+    xcontext 'when query_log_tags_enabled is false' do
+      before do
+        config = Rails.configuration.active_record
+        allow(config).to receive(:query_log_tags_enabled).and_return(false)
```

Comment:
I think what's happening here is that the combustion application is already loaded by the time we try to stub this, which means the railtie has already run. We might be able to do this if we explicitly call `Combustion.initialize!` in this `before` block rather than requiring calling it in `spec_helper.rb` (though there may be a better way than this)

Reply:
I managed to work around it by using `Rails.env.development?` as the condition instead.

Lot less prone to failure.